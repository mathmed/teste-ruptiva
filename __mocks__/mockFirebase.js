import * as admin from 'firebase';

const firebase = admin.firestore();
let set = jest.fn();
let get = jest.fn(() => [
  {
    data: () => ({
      name: 'Mateus',
      document: '123',
      type: 'individual',
      id: 1,
    }),
  },
]);
let doc = jest.fn(() => ({ set, delete: jest.fn() }));
jest.spyOn(firebase, 'collection').mockReturnValue({ doc });
jest.spyOn(firebase, 'collection').mockReturnValue({ get });
jest.spyOn(console, 'warn').mockImplementation(() => {});
jest.spyOn(console, 'error').mockImplementation(() => {});
