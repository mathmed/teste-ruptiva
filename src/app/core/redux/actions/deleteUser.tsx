import db from '../../services/firebase';
import { Dispatch } from '../store/types';

const deleteUser = (uuid: string) => async (
  dispatch: Dispatch,
): Promise<void> => {
  try {
    await db.collection('submissions').doc(uuid).delete();
  } catch (error) {}
};

export default deleteUser;
