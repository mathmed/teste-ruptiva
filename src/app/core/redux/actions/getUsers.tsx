import db from '../../services/firebase';
import {
  Dispatch,
  START_GET_USERS,
  FINISH_GET_USERS,
  UserData,
} from '../store/types';

const getUsers = () => async (dispatch: Dispatch): Promise<void> => {
  try {
    dispatch({ type: START_GET_USERS });
    const data: UserData[] = [];
    const users = await db.collection('submissions').get();
    users.forEach((user) => data.push(user.data() as UserData));
    dispatch({ type: FINISH_GET_USERS, payload: data });
  } catch (error) {
    dispatch({ type: FINISH_GET_USERS });
  }
};

export default getUsers;
