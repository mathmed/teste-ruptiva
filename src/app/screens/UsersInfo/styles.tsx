import styled from 'styled-components/native';

export const Container = styled.Modal``;

export const Title = styled.Text`
  color: #fff;
  font-weight: bold;
  text-transform: uppercase;
  font-weight: bold;
  text-align: center;
  font-size: 24px;
  padding-top: 10%;
`;

export const TextCloseModal = styled.Text`
  font-weight: bold;
  font-size: 18px;
  color: #312e38;
`;

export const Content = styled.View`
  position: relative;
  display: flex;
  height: 100%;
`;

export const DataContent = styled.View`
  margin: 20px;
  flex: 1;
`;
export const CloseModalContent = styled.View`
  width: 100%;
  position: absolute;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
`;

export const ContainerUserInfo = styled.View`
  background: #312e38;
  margin: 6px 0;
  padding: 20px;
  border-radius: 6px;
  flex-direction: row;
`;

export const Name = styled.Text`
  font-size: 20px;
  font-weight: bold;
`;

export const DocumentType = styled.Text`
  font-size: 16px;
  font-weight: bold;
  margin-top: 6px;
  color: #ff9000;
`;

export const UserInfoContent = styled.View`
  flex: 1;
  display: flex;
  justify-content: center;
`;
export const DeleteContent = styled.TouchableOpacity`
  display: flex;
  justify-content: center;
  align-items: flex-end;
`;
