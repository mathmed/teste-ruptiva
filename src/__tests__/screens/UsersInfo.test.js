import React from 'react';
import configureStore from 'redux-mock-store';
import ReduxThunk from 'redux-thunk';
import UsersInfo from '../../app/screens/UsersInfo';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '../../../__mocks__/mockFirebase';
import getUsers from '../../app/core/redux/actions/getUsers';
import deleteUser from '../../app/core/redux/actions/deleteUser';

let screen = null;
jest.useFakeTimers();
jest.setTimeout(30000);

describe('[SCREENS] UsersInfo', () => {
  const mockStore = configureStore([ReduxThunk]);
  const reducerInitialState = {
    loadingGetUsersRequest: false,
    loadingRegisterRequest: false,
    users: [],
  };
  const store = mockStore(reducerInitialState);

  beforeEach(() => {
    screen = renderer.create(
      <Provider store={store}>
        <UsersInfo />
      </Provider>,
    );
  });

  it('Should render corretly', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should calls getUsers action and returns data success', async () => {
    await store.dispatch(getUsers());
    const actions = store.getActions();
    expect.assertions(3);
    expect(actions[0].type).toEqual('START_GET_USERS');
    expect(actions[1].type).toEqual('FINISH_GET_USERS');
    expect(actions[1].payload).toEqual([
      {
        name: 'Mateus',
        document: '123',
        type: 'individual',
        id: 1,
      },
    ]);
  });

  it('Should calls deleteUser action and returns data success', async () => {
    await store.dispatch(deleteUser(10));
    const actions = store.getActions();
    expect.assertions(3);
    expect(actions[0].type).toEqual('START_GET_USERS');
    expect(actions[1].type).toEqual('FINISH_GET_USERS');
    expect(actions[1].payload).toEqual([
      {
        name: 'Mateus',
        document: '123',
        type: 'individual',
        id: 1,
      },
    ]);
  });
});
