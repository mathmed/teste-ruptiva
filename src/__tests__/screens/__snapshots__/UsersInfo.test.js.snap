// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SCREENS] UsersInfo Should render corretly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  presentationStyle="overFullScreen"
  style={
    Array [
      Object {},
    ]
  }
  transparent={false}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "height": "100%",
          "position": "relative",
        },
      ]
    }
  >
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <RCTScrollView
        bounces={false}
        contentContainerStyle={
          Object {
            "flex": 1,
          }
        }
        contentInsetAdjustmentBehavior="never"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000000AA",
                  "flex": 1,
                  "padding": 24,
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "width": "100%",
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e8e8e8",
                    "borderRadius": 5,
                    "elevation": 4,
                    "shadowOffset": Object {
                      "height": 4,
                      "width": 2,
                    },
                    "shadowOpacity": 0.24,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000DD",
                      "fontSize": 20,
                      "margin": 24,
                      "marginBottom": 0,
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Atenção
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "padding": 24,
                      "paddingTop": 20,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#00000089",
                        "fontSize": 18,
                        "textAlign": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  Quer realmente deletar esse registro?
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {},
                    undefined,
                  ]
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#00000011",
                        "height": 1,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "height": 46,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#0000FF99",
                              "fontWeight": "normal",
                              "textAlign": "center",
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Não
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "#00000011",
                          "width": 1,
                        }
                      }
                    />
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "height": 46,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#0000FF99",
                              "fontWeight": "bold",
                              "textAlign": "center",
                              "textAlignVertical": "center",
                            }
                          }
                        >
                          Sim
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </Modal>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 24,
            "fontWeight": "bold",
            "paddingTop": "10%",
            "textAlign": "center",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Cadastros
    </Text>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
          },
        ]
      }
    >
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            colors={
              Array [
                "#fff",
              ]
            }
            onRefresh={[Function]}
            progressBackgroundColor="#ff9000"
          />
        }
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <RCTRefreshControl />
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 0,
            "display": "flex",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "position": "absolute",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ff9000",
            "borderRadius": 10,
            "display": "flex",
            "height": 56,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#312e38",
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          FECHAR
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
